testing/kuttl/scripts/pgbackrest-initialization.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
```

Comment:
:thinking: I'd like to add some additional checking against the `pgbackrest info` output. One thought I had was to validate the correct number and  type of backups exist, i.e. `pgbackrest-initialization.sh` could be
```
#!/bin/bash

EXPECTED_STATUS=$1
EXPECTED_NUM_BACKUPS=$2

CLUSTER=${CLUSTER:-default}

REPO_POD=$(kubectl -n ${NAMESPACE} get pod --selector=postgres-operator.crunchydata.com/cluster=${CLUSTER},postgres-operator.crunchydata.com/pgbackrest-dedicated --no-headers -o custom-columns=":metadata.name")

# Grab the `status` line from `pgbackrest info` (regular output, not json since that output includes unnecessary info)
# remove whitespace with `xargs` and trim the string to only include the status
STATUS=$(kubectl -n ${NAMESPACE} exec -i ${REPO_POD} -c pgbackrest -- pgbackrest info |
    grep status |
    xargs |
    cut -d' ' -f 2)

if [[ "$STATUS" != "$EXPECTED_STATUS" ]]; then
    echo "Expected ${EXPECTED_STATUS} but got ${STATUS}"
    exit 1
fi

# Validate that the expected number of backups are found.
NUM_BACKUPS=$(kubectl -n ${NAMESPACE} exec -i ${REPO_POD} -c pgbackrest -- pgbackrest info |
     grep "full backup:" |
     wc -l)

if [[ "$NUM_BACKUPS" != "$EXPECTED_NUM_BACKUPS" ]]; then
    echo "Expected ${EXPECTED_NUM_BACKUPS} but got ${NUM_BACKUPS}"
    exit 1
fi
```

with the "script" line of `01-pgbackrest-connect.yaml` becoming
```
- script: CLUSTER=init-pgbackrest ../../scripts/pgbackrest-initialization.sh "mixed" 1
```
and `03-pgbackrest-connect.yaml` becoming
```
- script: CLUSTER=init-pgbackrest ../../scripts/pgbackrest-initialization.sh "ok" 2
```

Granted, the "mixed"/"ok" check is _sort of_ doing this already, but this is a bit more explicit.

Reply:
Took this suggestion in the update.